tiledb/sm/filesystem/s3.cc

```diff
@@ -224,6 +224,9 @@ Status S3::init(const Config& config, ThreadPool* const thread_pool) {
   auto aws_secret_access_key =
       config.get("vfs.s3.aws_secret_access_key", &found);
   assert(found);
+  auto aws_session_token =
+      config.get("vfs.s3.aws_session_token", &found);
+  assert(found);
```

Comment:
The commit message leads me to believe that tokens are not _always_ required. Should we really have an `assert` here?

Reply:
Consistency with the others -- the dict entry is also still expected for the key and secret, even if empty.

Reply:
ah, thanks!